src/Npgsql/PostgresException.cs

```diff
@@ -236,6 +236,40 @@ internal PostgresException(ReadBuffer buf)
         public override string Message => SqlState + ": " + MessageText;
 
         /// <summary>
+        /// Gets whether or not the exception can be treated as transient based on the SqlState. Useful when defining retry policies.
+        /// </summary>
+        public override bool IsTransient
+        {
+            get
+            {
+                switch (SqlState)
+                {   
+                    case "53000":   //insufficient_resources
+                    case "53100":   //disk_full
+                    case "53200":   //out_of_memory
+                    case "53300":   //too_many_connections
+                    case "53400":   //configuration_limit_exceeded
+                    case "57P03":   //cannot_connect_now
+                    case "58000":   //system_error
+                    case "58030":   //io_error
+                    case "40001":   //serialization_error
+                    case "55P03":   //lock_not_available
```

Comment:
My reasoning for leaving them in was essentially this response from the postgres hackers group: https://www.postgresql.org/message-id/CAMsr%2BYG3VkoV4P2SAqi16PeBMb8WBKi_FG7p0jFktz1rd4n1cg%40mail.gmail.com